### PR TITLE
Separate vsphere template resolver package

### PR DESCRIPTION
**Pull request by hongtaiwang**
### What this PR does / why we need it
remove vsphere template resolver from vsphere clusterclass package since we are going to separate clusterclass package with tanzu-framework.

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### Additional information

tested locally. pi for vsphere-resolver will be installed and the component works fine.
pi will not install in aws. Only package resources exist. working as expected

### CODE REVIEW

File: packages/vsphere-template-resolver/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# tkg-clusterclass-vsphere Package
```
Comment (imikushin) [inline packages/vsphere-template-resolver/README.md:1]: Did you mean "vsphere-template-resolver Package"?
Also, the text below still refers to tkg-clusterclass-vsphere.

File: packages/vsphere-template-resolver/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: PackageMetadata
+metadata:
+  name: vsphere-template-resolver.tanzu.vmware.com
+  namespace: vsphere-template-resolver
+spec:
+  displayName: "vsphere-template-resolver"
+  longDescription: "Vsphere-template-resolver resolves machine template based on os version and arch for vsphere provider"
```
Comment (imikushin) [inline packages/vsphere-template-resolver/metadata.yaml:8]:
```suggestion
  longDescription: "vSphere Template Resolver finds VM template based on OS version and arch for vSphere provider"
```

File: packages/vsphere-template-resolver/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  shortDescription: "Vsphere machine template resolver"
```
Comment (imikushin) [inline packages/vsphere-template-resolver/metadata.yaml:9]:
```suggestion
  shortDescription: "vSphere VM template resolver"
```

**Review by imikushin — APPROVED** [review]
LGTM. Thanks Hongtai!

Please address the comments (minor and easy to fix) before merging, though.

File: packages/vsphere-template-resolver/metadata.yaml (hunk 2 above)
Comment (hongtaiwang) [reply packages/vsphere-template-resolver/metadata.yaml]: done

File: packages/vsphere-template-resolver/metadata.yaml (hunk 3 above)
Comment (hongtaiwang) [reply packages/vsphere-template-resolver/metadata.yaml]: done

File: packages/vsphere-template-resolver/README.md (hunk 1 above)
Comment (hongtaiwang) [reply packages/vsphere-template-resolver/README.md]: ahhh  my bad. corrected the typo

File: packages/tkg/bundle/config/packageinstalls/tkr-vsphere-resolver-service-account.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: tkr-vsphere-resolver-package-role
+  annotations:
+    kapp.k14s.io/change-group: "tkr-vsphere-resolver-packageinstall/serviceaccount-0"
+rules:
+  #  RBAC rules to create PackageInstall CR and service accounts
+  - apiGroups: ["*"]
```
Comment (DanielXiao) [inline packages/tkg/bundle/config/packageinstalls/tkr-vsphere-resolver-service-account.yaml:21]: Is this too large permission for the vm template resolver?

**Comment by DanielXiao** [comment]
Please also need to remove roles which were added to `packages/tkg-clusterclass/bundle/config/packageinstalls/tkg-infra-clusterclass-service-account.yaml` in https://github.com/vmware-tanzu/tanzu-framework/pull/3683.

File: packages/tkg/bundle/config/packageinstalls/tkr-vsphere-resolver-service-account.yaml (hunk 4 above)
Comment (hongtaiwang) [reply packages/tkg/bundle/config/packageinstalls/tkr-vsphere-resolver-service-account.yaml:21]: it is to keep consistent with all other pi rbac within this folder

**Comment by hongtaiwang** [comment]
> Please also need to remove roles which were added to `packages/tkg-clusterclass/bundle/config/packageinstalls/tkg-infra-clusterclass-service-account.yaml` in #3683.

we are going to remove tkg-clusterclass package in future release, do we still need to modify those rbacs? It will cost some time to run mr pipeline if I push another commit to remove those.

**Pull request merged**